File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-3.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> solution;
        stack<TreeNode*> stac;
        TreeNode* curr = root;

        while (curr != nullptr || !stac.empty()) {
            while (curr != nullptr) {
                stac.push(curr);
                curr = curr->left;
            }
            curr = stac.top();
            stac.pop();
            solution.push_back(curr->val);
            curr = curr->right;
        }
        return solution;

    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-3.py
class Solution:
    def isValid(self, s: str) -> bool:
        validBrackets = {'(':')', '{':'}', '[':']'}
        stack = []

        for char in s:
            if stack and stack[-1] in validBrackets and char == validBrackets[stack[-1]]:
                stack.pop()
            else:
                stack.append(char)  

        if stack:
            return False
        return True


        